extensions/package-vulnerability-scanner/src/components/ContentList.vue

```diff
@@ -4,11 +4,16 @@ import { computed, ref } from "vue";
 import { usePackagesStore } from "../stores/packages";
 import { useContentStore } from "../stores/content";
 import { useScannerStore } from "../stores/scanner";
+import type { User } from "../stores/user";
 import StatusMessage from "./ui/StatusMessage.vue";
 import SkeletonText from "./ui/SkeletonText.vue";
 import BadgeTabs, { type Tab } from "./ui/BadgeTabs.vue";
 import ContentCard from "./ContentCard.vue";
 
+const props = defineProps<{
+  user: User;
+}>();
```

Comment:
Since `ContentList` already makes use of stores, is there a reason to create a prop and not import `userStore` directly here?

Reply:
I do this to help with typing.

`App.vue` has a loading spinner while the vulnerability store, content store, and user store are loading in data. The `ContentList` only renders when the user store has a user, so if I pass it as a prop I can tell this component that it is guaranteed to have the data. If I use `userStore.user` directly I will have to either have `undefined` checks or use the TypeScript non-null assertion operator (`!`)